src/java/org/apache/cassandra/exceptions/RequestFailureReason.java

```diff
@@ -53,26 +59,31 @@ public enum RequestFailureReason
         this.code = code;
     }
 
-    private static final RequestFailureReason[] codeToReasonMap;
+    private static final Map<Integer, RequestFailureReason> codeToReasonMap = new HashMap<>();
+    private static final Map<Class<? extends Throwable>, RequestFailureReason> exceptionToReasonMap = new HashMap<>();
 
     static
     {
         RequestFailureReason[] reasons = values();
 
-        int max = -1;
-        for (RequestFailureReason r : reasons)
-            max = max(r.code, max);
-
-        RequestFailureReason[] codeMap = new RequestFailureReason[max + 1];
-
         for (RequestFailureReason reason : reasons)
         {
-            if (codeMap[reason.code] != null)
+            if (codeToReasonMap.put(reason.code, reason) != null)
                 throw new RuntimeException("Two RequestFailureReason-s that map to the same code: " + reason.code);
```

Comment:
```suggestion
                throw new RuntimeException("Two RequestFailureReasons that map to the same code: " + reason.code);
```
Not sure what the dash was doing in there...

Reply:
typo? or someone wanted to stress on the class name?

Reply:
Do you want me to revert it?

Reply:
I guess if we're here, we might as well get rid of the dash.